distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java

```diff
@@ -270,65 +292,84 @@ public int getHitCount(Query query) throws Exception {
 
-    public void retrieveProductFromDAG(DAG dag) throws MalformedURLException {
+    public void downloadProductFromDAG(DAG dag) throws MalformedURLException {
         LOGGER.info("Downloading products...");
-        for (int i = 0; i < dag.nodes.length; i++) {
-            Node node = dag.nodes[i];
-            if (node.attribute_name.equals("productUrl")) {
-
-                String url = node.value.extract_string();
-                URL fileDownload = new URL(url);
-                String productPath = "product.jpg";
-                LOGGER.info("Downloading product : {}", url);
+
+        for (Node node : dag.nodes) {
+            if (node.attribute_name.equals(NsiliConstants.PRODUCT_URL)) {
+                URL fileDownloadUrl = getEncodedUrlFromString(node.value.extract_string());
+
+                final String productPath = "product.jpg";
+                LOGGER.info("Downloading product : {}", fileDownloadUrl);
                 try (FileOutputStream outputStream = new FileOutputStream(new File(productPath));
-                        BufferedInputStream inputStream = new BufferedInputStream(fileDownload.openStream())) {
+                        BufferedInputStream inputStream = new BufferedInputStream(fileDownloadUrl.openStream())) {
 
                     byte[] data = new byte[1024];
                     int count;
                     while ((count = inputStream.read(data, 0, 1024)) != -1) {
                         outputStream.write(data, 0, count);
                     }
 
-                    LOGGER.info("Successfully downloaded product from {}.", url);
+                    LOGGER.info("Successfully downloaded product from {}.", fileDownloadUrl);
                     Files.deleteIfExists(Paths.get(productPath));
-
                 } catch (IOException e) {
-                    LOGGER.error("Unable to download product from {}.", url, e);
+                    LOGGER.error("Unable to download product from {}.", fileDownloadUrl, e);
                 }
             }
         }
     }
 
-    public PackageElement[] order(ORB orb, DAG[] dags) throws Exception {
+    private URL getEncodedUrlFromString(String urlString) throws MalformedURLException {
+        URL url = new URL(urlString);
+        try {
+            URI uri = new URI(url.getProtocol(),
+                    url.getUserInfo(),
+                    url.getHost(),
+                    url.getPort(),
+                    url.getPath(),
+                    url.getQuery(),
+                    url.getRef());
+            url = uri.toURL();
+        } catch (URISyntaxException e) {
+            LOGGER.error("Unable to encode file download URL {}", url, e);
+        }
+        return url;
+    }
+
+    /**
+     * @param dag
+     * @return the first file name from the order response
+     * @throws Exception
+     */
+    public String testOrder(DAG dag) throws Exception {
+        PackageElement[] packageElements = order(dag);
+        return packageElements[0].files[0];
```

Comment:
This method is only called in `NsiliEndpointTest`, and we would expect neither array to be empty after having ingested one record.